sonar-php-plugin/src/test/java/org/sonar/plugins/php/phpunit/xml/TestSuitesTest.java

```diff
@@ -22,78 +22,37 @@
  */
 package org.sonar.plugins.php.phpunit.xml;
 
-import org.junit.Before;
+import java.util.List;
 import org.junit.Test;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
-
-import java.lang.reflect.Constructor;
-import java.util.ArrayList;
-import java.util.List;
-import java.util.TreeSet;
+import org.sonar.plugins.php.phpunit.PhpUnitTestFileReport;
 
 import static org.assertj.core.api.Assertions.assertThat;
-import static org.junit.Assert.assertSame;
-import static org.junit.Assert.assertThat;
-import static org.junit.matchers.JUnitMatchers.both;
-import static org.junit.matchers.JUnitMatchers.hasItem;
+import static org.junit.Assert.assertTrue;
 
-/**
- * Test for TestSuites.
- *
- * @author gennadiyl
- */
 public class TestSuitesTest {
 
-  private TestSuites testSuites;
-  private static final Logger LOG = LoggerFactory.getLogger(TestSuitesTest.class);
-
-  /**
-   * @throws java.lang.Exception
-   */
-  @Before
-  public void setUp() throws Exception {
-    testSuites = new TestSuites();
-  }
-
   /**
    * Empty constructors are required by xstream for TestSuite and TestSuite, in order to
    * be compatible with Java 7.
    */
   @Test
   public void test_compatible_java_7() {
-    Constructor[] constructors1 = getConstructorForClass("org.sonar.plugins.php.phpunit.xml.TestSuite");
-    assertThat(constructors1).isNotEmpty();
-    assertThat(getNbParamPerConstructor(constructors1).first()).isEqualTo(0);
-
-    Constructor[] constructors2 = getConstructorForClass("org.sonar.plugins.php.phpunit.xml.TestSuites");
-    assertThat(constructors2).isNotEmpty();
-    assertThat(getNbParamPerConstructor(constructors2).first()).isEqualTo(0);
-  }
-
-  private Constructor[] getConstructorForClass(String className) {
-    Constructor[] constructors = new Constructor[0];
-    try {
-      Class testSuites = Class.forName(className);
-      constructors = testSuites.getConstructors();
-    } catch (ClassNotFoundException e) {
-      LOG.error("If TestSuite and TestSuites class does not exist any more, this test might be useless.", e);
-    }
-    return constructors;
+    new TestSuites();
+    new TestSuite();
```

Comment:
This was originally a very convoluted test using reflection to validate that both TestSuite and TestSuites had zero-argument constructors. I replaced it with these two instantiations. It seemed a bit overkill to also create a new test case in TestSuiteTest just to do new TestSuite. The idea is that, as the need arises, we can list here all the classes where we want to enforce the presence of the zero-argument constructor. Now that I think of it... I'll do a new of all the xml/ classes here.